src/cmds/net/telnetd/telnetd.c

```diff
@@ -169,16 +168,14 @@ static void *shell_hnd(void* args) {
 		MD(printf("dup2 STDERR_FILENO error: %d\n", errno));
 	}
 
-	ret = shell_run(shell_lookup("tish"));
-	if (ret != 0) {
-		MD(printf("shell_run error: %d\n", ret));
-	}
+	system("tish");
```

Comment:
Now system() implemented as single call in the same thread for reducing recourses. 
But you're right fork+exec+waitpid is better way than now.